src/drivers/mct/cime_config/config_component.xml

```diff
@@ -2400,10 +2400,10 @@
   <entry id="PIO_REARR_COMM_MAX_PEND_REQ_COMP2IO">
     <type>integer</type>
     <valid_values></valid_values>
-    <default_value>0</default_value>
+    <default_value>-2</default_value>
     <group>run_pio</group>
     <file>env_run.xml</file>
-    <desc>pio rearranger communication max pending requests (comp2io) : 0 implies that CIME internally calculates the value ( = max(64, 2 * PIO_NUMTASKS) ), -1 implies no bound on max pending requests </desc>
+    <desc>pio rearranger communication max pending requests (comp2io) : -2 implies that CIME internally calculates the value ( = max(64, 2 * PIO_NUMTASKS) ), -1 implies no bound on max pending requests </desc>
```

Comment:
Should add here that 0 implies use MPI_ALLTOALL

Reply:
There is another variable, PIO_REARR_COMM_TYPE, that was intended to control whether the user wants a collective or point to point messages for data rearrangement.

Reply:
So when I submitted PR #3499 which you reviewed and approved - why was this not mentioned?  The implication is that #3499 should have set PIO_REARR_COMM_TYPE instead.

Reply:
I haven't looked at the PIO code in a while, I assumed that you wanted to use the maximum pending requests setting to force the collective usage in PIO. In any case, a value of 0 can be considered a valid setting for the maximum pending requests and we need to move to a different value for the default setting (to indicate CIME calculate the value for the user)

Reply:
Perhaps making that change to #3499 would remove the need for this PR?

Reply:
AFAICT, this behavior (using the comm type to choose the behavior) would also require change in PIO. In any case, this PR allows users to set maximum pending requests = 0, if needed.